files/tk_metrics

```diff
@@ -7,44 +7,71 @@ require 'time'
 require 'optparse'
 require 'yaml'
 
-options = {}
+#===========================================================================#
+# BUILD CONFIGURATION OPTIONS                                               #
+#===========================================================================#
+
+# Define the list of mandatory options that must be specified on the CLI
+REQUIRED_FLAGS = {
+  :"metrics-type" => { type: :string },
+}
+
+# Define the list of options that will be read from the config file, and which
+# may optionally be specified on the CLI.
+SETTINGS = {
+  :"output-dir"   => { type: :string, default: false }, # false = disabled
+  :hosts          => { type: :list },
+  :metrics        => { type: :list },
+  :"metrics-port" => { type: :string },
+  :clientcert     => { type: :string },
+  :"pe-version"   => { type: :string },
+  :print          => { type: :boolean, default: true },
+  :ssl            => { type: :boolean, default: true },
+}
+
+config = { }
+
 OptionParser.new do |opts|
   opts.banner = "Usage: tk_metrics [options]"
 
-  opts.on('-p', '--[no-]print', 'Print to stdout') { |p| options[:print] = p }
-  opts.on('-m [TYPE]', '--metrics_type [TYPE]', 'Type of metric to collect') { |v| options[:metrics_type] = v }
-  opts.on('-o [DIR]', '--output-dir [DIR]', 'Directory to save output to') { |o| options[:output_dir] = o }
-  opts.on('--metrics_port [PORT]', 'The port the metrics service runs on') { |port| options[:metrics_port] = port }
-  opts.on('--[no-]ssl', 'Whether or not to use SSL when gather metrics') { |ssl| options[:ssl] = ssl }
+  # Gather options
+  REQUIRED_FLAGS.merge(SETTINGS).each do |opt,attrs|
+    case attrs[:type]
+    when :string, nil
+      opts.on("--#{opt} ARG", '(no description)')  { |arg| config[opt] = arg }
```

Comment:
They can be, yeah. This comment prompted a semi-big rewrite of the consolidation. New commits pushed.